crates/forge/tests/integration/set_balance.rs

```diff
@@ -0,0 +1,244 @@
+            }
+        "#
+        ),
+        Contract::from_code_path(
+            "ERC20".to_string(),
+            Path::new("tests/data/contracts/erc20.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run_test_case(&test, ForgeTrackedResource::CairoSteps);
+
+    assert_passed(&result);
+}
+
+#[test]
+fn test_set_balance_big_amount() {
+    let test = test_case!(
+        indoc!(
+            r#"
+            use core::num::traits::Pow;
+            use snforge_std::TokenTrait;
+            use starknet::{ContractAddress, syscalls, SyscallResultTrait};
+
+            use snforge_std::{set_balance, Token};
+
+            fn get_balance(contract_address: ContractAddress, token: Token) -> Span<felt252> {
+                let mut calldata: Array<felt252> = array![contract_address.into()];
+                let balance = syscalls::call_contract_syscall(
+                    token.contract_address(), selector!("balance_of"), calldata.span(),
+                )
+                    .unwrap_syscall();
+                balance
+            }
+
+            #[test]
+            fn test_set_balance_big_amount() {
+                let contract_address: ContractAddress = 0x123.try_into().unwrap();
+
+                let balance_before = get_balance(contract_address, Token::STRK);
+                assert_eq!(balance_before, array![0, 0].span(), "Balance should be 0");
+
+                set_balance(contract_address, (10.pow(50_u32)).try_into().unwrap(), Token::STRK);
+
+                let balance_after = get_balance(contract_address, Token::STRK);
+                assert_eq!(
+                    balance_after,
+                    array![194599656488044247630319707454198251520, 293873587705].span(),
+                    "Balance should should be 10^50",
+                );
+            }
+        "#
+        ),
+        Contract::from_code_path(
+            "HelloStarknet".to_string(),
+            Path::new("tests/data/simple_package/src/hello_starknet.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run_test_case(&test, ForgeTrackedResource::CairoSteps);
+
+    assert_passed(&result);
+}
+
+#[test]
+fn test_set_balance_strk_with_fork() {
+    let test = test_case!(
+        indoc!(
+            r#"
+            use snforge_std::TokenTrait;
+            use starknet::{ContractAddress, syscalls, SyscallResultTrait};
+
+            use snforge_std::{set_balance, Token};
+
+            fn get_balance(contract_address: ContractAddress, token: Token) -> Span<felt252> {
+                let mut calldata: Array<felt252> = array![contract_address.into()];
+                let balance = syscalls::call_contract_syscall(
+                    token.contract_address(), selector!("balance_of"), calldata.span(),
+                )
+                    .unwrap_syscall();
+                balance
+            }
+
+            #[fork(url: "http://188.34.188.184:7070/rpc/v0_8", block_number: 715_593)]
```

Comment:
See `setup_fork.rs`, you can use `formatdoc!` and inject the value of `node_rpc_url()`